src/dodal/devices/i10/id_apple2.py

```diff
@@ -0,0 +1,478 @@
+import asyncio
+import csv
+from pathlib import Path
+
+import numpy as np
+from bluesky.protocols import Movable
+from ophyd_async.core import (
+    AsyncStatus,
+    HintedSignal,
+    StandardReadable,
+    soft_signal_r_and_setter,
+    soft_signal_rw,
+    wait_for_value,
+)
+
+from dodal.devices.apple2_undulator import (
+    Apple2Val,
+    UndlatorPhaseAxes,
+    UndulatorGap,
+    UndulatorGateStatus,
+)
+from dodal.devices.pgm import PGM
+from dodal.log import LOGGER
+
+ROW_PHASE_MOTOR_TOLERANCE = 0.004
+MAXIMUM_ROW_PHASE_MOTOR_POSITION = 24.0
+MAXIMUM_GAP_MOTOR_POSITION = 100
+
+
+class I10Apple2(StandardReadable, Movable):
+    """
+    Apple 2 ID/undulator has 4 extra degrees of freedom compare to the standard Undulator,
+     each bank of magnet can move independently to each other,
+     which allow the production of different x-ray polarisation as well as energy.
+     This type of ID is use on I10, I21, I09, I17 and I06 for soft x-ray.
```

Comment:
if it's in use at so many beamlines, it could be a shared device possibly (?) it's not like those beamlines are soon to go to bluesky afaik though, so it's fine